File: vortexdbg-api/src/main/java/com/vortexdbg/unix/struct/TimeVal32.java
package com.vortexdbg.unix.struct;

import com.vortexdbg.pointer.VortexdbgStructure;
import com.sun.jna.Pointer;

import java.util.Arrays;
import java.util.List;

public class TimeVal32 extends VortexdbgStructure {

    public TimeVal32(Pointer p) {
        super(p);
    }

    public int tv_sec;
    public int tv_usec;

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("tv_sec", "tv_usec");
    }

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/unix/struct/TimeVal32.kt
package com.vortexdbg.unix.struct

import com.vortexdbg.pointer.VortexdbgStructure
import com.sun.jna.Pointer

class TimeVal32(p: Pointer?) : VortexdbgStructure(p) {

    @JvmField
    var tv_sec: Int = 0
    @JvmField
    var tv_usec: Int = 0

    override fun getFieldOrder(): List<String> {
        return listOf("tv_sec", "tv_usec")
    }
}
